IdentityCore/src/webview/MSIDWebviewAuthorization.m

```diff
@@ -26,33 +26,114 @@
-                                                                     factory:(MSIDOAuth2Factory *)factory;
++ (void)startEmbeddedWebviewWebviewAuthWithRequestParameters:(MSIDRequestParameters *)parameters
+                                                     webview:(WKWebView *)webview
+                                                     factory:(MSIDOauth2Factory *)factory
+                                                     context:(id<MSIDRequestContext>)context
+                                           completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
-    return nil;
+    id<MSIDWebviewInteracting> embeddedWebviewController = [factory embeddedWebviewControllerWithRequest:parameters
+                                                                                           customWebview:webview
+                                                                                       completionHandler:completionHandler];
+    [self startWebviewAuth:embeddedWebviewController
+                   context:context
+         completionHandler:completionHandler];
 }
 
-+ (id<MSIDWebviewInteracting>)embeddedWebviewControllerWithRequestParameters:(MSIDRequestParameters *)parameters
-                                                                     webview:(WKWebView *)webview
-                                                                     factory:(MSIDOAuth2Factory *)factory;
++ (void)startSystemWebviewWebviewAuthWithRequestParameters:(MSIDRequestParameters *)parameters
+                                         callbackURLScheme:(NSString *)callbackURLScheme
+                                                   factory:(MSIDOauth2Factory *)factory
+                                                   context:(id<MSIDRequestContext>)context
+                                         completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
-    return nil;
+    id<MSIDWebviewInteracting> systemWebviewController = [factory systemWebviewControllerWithRequest:parameters
+                                                                                   callbackURLScheme:callbackURLScheme
+                                                                                   completionHandler:completionHandler];
+    [self startWebviewAuth:systemWebviewController
+                   context:context
+         completionHandler:completionHandler];
 }
 
-+ (id<MSIDWebviewInteracting>)systemWebviewControllerWithRequestParameters:(MSIDRequestParameters *)parameters
-                                                                   factory:(MSIDOAuth2Factory *)factory;
++ (void)startWebviewAuth:(id<MSIDWebviewInteracting>)webviewController
+                 context:(id<MSIDRequestContext>)context
+       completionHandler:(MSIDWebUICompletionHandler)completionHandler
 {
-    return nil;
+    if (![self setCurrentWebSession:webviewController])
```

Comment:
nit: should we check this before creating a controller?

Reply:
Thought about that too, it seems just a little more logical this way although it might be less efficient.